app/src/main/java/com/yacgroup/yacguide/TourbookActivity.kt

```diff
@@ -311,17 +326,35 @@ class TourbookActivity : BaseNavigationActivity() {
     private fun _selectFileImport() {
         val intent = Intent(Intent.ACTION_OPEN_DOCUMENT).apply {
             addCategory(Intent.CATEGORY_OPENABLE)
-            type = "*/*"
+            type = MIME_TYPE_JSON
         }
-        startActivityForResult(intent, IntentConstants.REQUEST_OPEN_TOURBOOK)
+        _importResultLauncher.launch(intent)
+    }
+
+    private fun _selectExportFormat() {
+        var mimeType = MIME_TYPE_JSON
+        DialogWidgetBuilder(this, R.string.export_format).apply {
+            setSingleChoiceItems(R.array.exportFormats, 0) {_, which ->
+                when (which) {
+                    0 -> _tourbookExporter.exportFormat = TourbookExporter.ExportFormat.eJSON
```

Comment:
I guess, the cleanest way is to make TourbookExporter abstract and have dedicated `JSONTourbookExporter` and `CSVTourbookExporter`, respectively, as well as a dedicated importer class.
These concrete exporters would then handle their own export files types (mimetype).

However, this could be done as a follow-up if the scenario above works with the current approach.

Reply:
> Is this only executed when you actively select an item? Do we have a problem with the following:
> 
> 1. Open the dialog and select CSV (-> export format is set at exporter)
> 2. Cancel the dialog (negative button)
> 3. Open the dialog again and don't select anything (I guess, JSON (= idx 0). is default).
> 4. Confirm and choose your json file.
> 5. Now as we call `TourbookExporter.export()`, we want to export json but have we really set the export format or is it still CSV?
> 
> That's what I mean with "dangerous".

Good catch. Yes, it is as you described.

Reply:
> I guess, the cleanest way is to make TourbookExporter abstract and have dedicated `JSONTourbookExporter` and `CSVTourbookExporter`, respectively, as well as a dedicated importer class. These concrete exporters would then handle their own export files types (mimetype).
> 
> However, this could be done as a follow-up if the scenario above works with the current approach.

I fixed the issue in a more less nice way but agree whit your proposal.